File: Python/Programmers/Level 1/수박수박수박수박수박수?.py
# def solution(n):
#     answer = "수박"
#     if n % 2 == 0: # n이 짝수이면 n//2만큼을 곱합
#         answer *= n//2
#     else: # 홀수면 (n-1)//2만큼을 곱하고 answer의 첫번째 요소를 더함
#         answer = answer * ((n-1)//2) + answer[0]
        
#     return answer
  
# def solution(n):
#     pattern = ["수", "박"]
#     answer = ''
#     for i in range(n):
#         answer += pattern[i % 2]

#     return answer

# def solution(n):
#     return ("수박"*n)[:n] # 수박 문자열에 n을 곱하고 n만큼 split

def solution(n):
    # 위 코드는 n이 증가하면 불필요한 메모리를 차지하는 단점이 있음
    # (n + 1) // 2의 값만 곱하면 불필요한 메모리를 최소화하여 개선
    return ("수박" * ((n + 1) // 2))[:n]

File: JavaScript/Programmers/Level 1/수박수박수박수박수박수?.js
// n = 3인 경우 "수박수", n = 4인 경우 "수박수박"
// 즉, "수박"을 (n + 1) / 2 만큼 반복하고 n까지 slice하면 됨

// function solution(n) {
//     const pattern = "수박";
//     let answer = "";
//     for (let i = 0; i < (n + 1) / 2; i++) { // (n + 1) / 2만큼 반복
//         answer += pattern;
//     }
    
//     return answer.slice(0, n); // n까지 slice
// }

// repeat 메소드를 이용하여 문자열 반복

function solution(n) {
    return "수박".repeat((n + 1) / 2).slice(0, n); // repeat 메소드를 이용하여 (n + 1) / 2만큼 반복하고 n까지 slice
}
